src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -1152,22 +1152,38 @@ private String GetCodeForReadyToRunHeader(ReadyToRunHeaderNode headerNode, NodeF
             rtrHeader.AppendLine();
             rtrHeader.Append("static struct {");
             rtrHeader.AppendLine();
+#if BIT64
```

Comment:
We do not use ifdefs for bitness specific checks in the compiler. Instead, we use dynamic checks to make the compiler cross-targeting easier. Look for `Target.PointerSize` for examples on how to do it.